app/ui/src/main/java/com/fsck/k9/ui/settings/account/AccountSettingsFragment.kt

```diff
@@ -300,6 +323,31 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
         return viewModel.getAccountBlocking(accountUuid)
     }
 
+    private fun onDeleteAccount() {
+        val dialogFragment = ConfirmationDialogFragment.newInstance(
+                DIALOG_DELETE_ACCOUNT,
+                getString(R.string.account_delete_dlg_title),
+                getString(R.string.account_delete_dlg_instructions_fmt, getAccount().description),
+                getString(R.string.okay_action),
+                getString(R.string.cancel_action)
+        )
+        dialogFragment.setTargetFragment(this, REQUEST_DELETE_ACCOUNT)
+        dialogFragment.show(fragmentManager, TAG_DELETE_ACCOUNT_CONFIRMATION)
+    }
+
+    override fun doPositiveClick(dialogId: Int) {
```

Comment:
Not very fond of generic listener method names in a long class like this. Use an anonymous listener object maybe?

Reply:
I agree. But `ConfirmationDialogFragment` uses `Fragment.getTargetFragment()` for the listener. This has the advantage that we don't have to worry about setting the listener again after the Activity/Fragment has been destroyed and recreated.